MapboxDirectionsTests/IntructionsTests.swift

```diff
@@ -83,22 +83,37 @@ class SpokenInstructionsTests: XCTestCase {
         
-        let visualInstructions = step.instructionsDisplayedAlongStep
+        var visualInstructions = step.instructionsDisplayedAlongStep
+        let visualInstructionComponent = visualInstructions?.first?.primaryInstruction.components.first as! VisualInstructionComponent
         
         XCTAssertNotNil(visualInstructions)
+        XCTAssertNotNil(visualInstructionComponent)
         XCTAssertEqual(visualInstructions?.first?.primaryInstruction.text, "Page Street")
-        XCTAssertEqual(visualInstructions?.first?.primaryInstruction.textComponents.first!.text, "Page Street")
+        XCTAssertEqual(visualInstructionComponent.text, "Page Street")
         XCTAssertEqual(visualInstructions?.first?.distanceAlongStep, 1107.1)
         XCTAssertEqual(visualInstructions?.first?.primaryInstruction.finalHeading, 180.0)
         XCTAssertEqual(visualInstructions?.first?.primaryInstruction.maneuverType, .turn)
         XCTAssertEqual(visualInstructions?.first?.primaryInstruction.maneuverDirection, .left)
-        XCTAssertEqual(visualInstructions?.first?.primaryInstruction.textComponents.first?.type, .text)
-        XCTAssertEqual(visualInstructions?.first?.primaryInstruction.textComponents.first?.abbreviation, "Page St")
-        XCTAssertEqual(visualInstructions?.first?.primaryInstruction.textComponents.first?.abbreviationPriority, 0)
+        XCTAssertEqual(visualInstructionComponent.type, .text)
+        XCTAssertEqual(visualInstructionComponent.abbreviation, "Page St")
+        XCTAssertEqual(visualInstructionComponent.abbreviationPriority, 0)
         XCTAssertEqual(visualInstructions?.first?.drivingSide, .right)
         XCTAssertNil(visualInstructions?.first?.secondaryInstruction)
         
         let arrivalVisualInstructions = arrivalStep.instructionsDisplayedAlongStep!
         XCTAssertEqual(arrivalVisualInstructions.first?.secondaryInstruction?.text, "the gym")
+        
+        // Tertiary Visual Instructions
+        visualInstructions = leg.steps[5].instructionsDisplayedAlongStep
+        let laneIndicationComponent = visualInstructions?.first?.tertiaryInstruction?.components.first as! LaneIndicationComponent
+        XCTAssertNotNil(visualInstructions)
+        XCTAssertEqual(visualInstructions?.first?.tertiaryInstruction?.text, "")
+        XCTAssertEqual(visualInstructions?.first?.distanceAlongStep, 120.7)
+        XCTAssertEqual(visualInstructions?.first?.tertiaryInstruction?.finalHeading, 180.0)
+        XCTAssertEqual(visualInstructions?.first?.tertiaryInstruction?.maneuverType, .reachFork)
+        XCTAssertEqual(visualInstructions?.first?.tertiaryInstruction?.maneuverDirection, .right)
+        XCTAssertEqual(laneIndicationComponent.isUsable, true)
+        XCTAssertEqual(laneIndicationComponent.indications, [.left, .straightAhead])
+        XCTAssertEqual(visualInstructions?.first?.drivingSide, .right)
```

Comment:
So does this step have both lane information and then information? I think the implementation on the server prevents this from ever occurring.

Reply:
Will re-check the JSON response and relay my finding.